app/src/main/java/com/rajatgoyal/puzzle15/ui/BaseActivity.java

```diff
@@ -0,0 +1,152 @@
+package com.rajatgoyal.puzzle15.ui;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+import timber.log.Timber;
+
+import android.content.Intent;
+import android.os.Bundle;
+
+import com.google.android.gms.auth.api.signin.GoogleSignIn;
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+import com.google.android.gms.auth.api.signin.GoogleSignInClient;
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
+import com.google.android.gms.common.api.ApiException;
+import com.google.android.gms.games.AchievementsClient;
+import com.google.android.gms.games.Games;
+import com.google.android.gms.games.PlayersClient;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.rajatgoyal.puzzle15.R;
+import com.rajatgoyal.puzzle15.util.AchievementHandler;
+import com.rajatgoyal.puzzle15.util.SharedPref;
+
+public abstract class BaseActivity extends AppCompatActivity {
```

Comment:
Make this class package private as it is used only in this package

Reply:
Top level classes may be `public` or not, but `private` or `protected` are not allowed. So, I have removed the `public` keyword as this class is being used in this package only.